mysql/gtid.go

```diff
@@ -29,3 +29,8 @@ func ParseGTIDSet(flavor string, s string) (GTIDSet, error) {
 		return nil, errors.Errorf("invalid flavor %s", flavor)
 	}
 }
+
+type GTIDEvent interface {
+	GTIDNext() (GTIDSet, error)
+	Decode(data []byte) error
```

Comment:
How do we use `Decode`? I see in the two implementations of `GTIDEvent` `Decode` changes the receiver, but `GTIDEvent` is an interface we know nothing about the actual receiver without type cast. And if we use type cast, we can directly call `MySQLGTIDEvent.Decode`, no need to add it to the interface.